dotCMS/src/main/java/com/dotcms/storage/repository/HashedLocalFileRepositoryManager.java

```diff
@@ -0,0 +1,52 @@
+package com.dotcms.storage.repository;
+
+import com.google.common.annotations.VisibleForTesting;
+
+import java.io.File;
+
+/**
+ * This is pretty much the implementation of the LocalFileRepositoryManager, but is expecting a hash as a filename
+ * so will normalize it as an (first 4 letters will be the parent sub folder) and the rest will be the file name.
+ * @author jsanca
+ */
+public class HashedLocalFileRepositoryManager extends  LocalFileRepositoryManager {
+
+    public HashedLocalFileRepositoryManager() {
+        super();
+    }
+
+    @VisibleForTesting
+    protected HashedLocalFileRepositoryManager(final String basePath) {
+        super(basePath);
+    }
+
+    /**
+     * If this receives f813463c714e009df1227f706e290e01, it is converted to
```

Comment:
Don't you think it would be easier for maintenance and locate a file also using the path of the original asset here for the metadata. ? 
something like 
/0/1/01a0e73b-6344-4285-8146-0372b5982c99/fileAsset

That way the path is reminiscent of the original asset and we don't have to use a hash to locate it @jdotcms ?